File: my_restaurent/src/main/java/com/example/my_restaurent/Admin/AdminController.java
package com.example.my_restaurent.Admin;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/admin")
public class AdminController {

    private final AdminService adminService;

    public AdminController(AdminService adminService) {
        this.adminService = adminService;
    }

    @GetMapping("/create")
    public String showCreateForm(Admin admin) {
        return "admin-create";
    }

    @PostMapping("/save")
    public String save(Admin admin) {
        adminService.addNewAdmin(admin);
        return "redirect:/admin/list";
    }

    @GetMapping("/{id}/edit")
    public String showUpdateForm(@PathVariable("id") long id, Model model) {
        Admin admin = adminService.getAdmin(id);
        model.addAttribute("admin", admin);
        return "update-admin";
    }

    @PostMapping("/{id}/update")
    public String updateAdmin(@PathVariable("id") long id, Admin admin) {
        adminService.updateAdmin(id, admin.getName(), admin.getEmail(), admin.getPassword());
        return "redirect:/admin/list";
    }

    @GetMapping("/{id}/delete")
    public String deleteAdmin(@PathVariable("id") long id) {
        adminService.deleteAdmin(id);
        return "redirect:/admin/list";
    }
}

File: my_restaurant/src/main/kotlin/Admin/AdminController.kt
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/admin")
class AdminController(private val adminService: AdminService) {

    @GetMapping("/list")
    fun list(model: Model): String {
        model.addAttribute("admins", adminService.findAll())
        return "list"
    }

    @GetMapping("/{id}")
    fun getAdmin(model: Model, @PathVariable id: Long): String {
        model.addAttribute("admin", adminService.findById(id))
        return "admin"
    }

    @PostMapping("/add")
    fun add(admin: Admin): String {
        adminService.save(admin)
        return "redirect:list"
    }

    @PostMapping("/{id}/update")
    fun update(@PathVariable id: Long, admin: Admin): String {
        adminService.update(id, admin)
        return "redirect:../list"
    }

    @DeleteMapping("/{id}/delete")
    fun delete(@PathVariable id: Long): String {
        adminService.deleteById(id)
        return "redirect:../list"
    }

}